src/main/java/org/folio/fqm/repository/EntityTypeRepository.java

```diff
@@ -110,6 +137,63 @@ public void replaceEntityTypeDefinitions(List<EntityType> entityTypes) {
     });
   }
 
+  public Optional<EntityType> getCompositeEntityTypeDefinition(UUID entityTypeId) {
+    log.info("Getting COMPOSITE definition for entity type ID: {}", entityTypeId);
+    Field<String> definitionField = field("definition", String.class);
+
+    EntityType entityType = jooqContext
+      .select(definitionField)
+      .from(table(TABLE_NAME))
+      .where(field(ID_FIELD_NAME).eq(entityTypeId))
+      .fetchOptional(definitionField)
+      .map(this::unmarshallEntityType)
+      .map(currentEntityType -> {
+        String customFieldsEntityTypeId = currentEntityType.getCustomFieldEntityTypeId();
+        if (customFieldsEntityTypeId != null) {
+          currentEntityType.getColumns().addAll(fetchColumnNamesForCustomFields(UUID.fromString(customFieldsEntityTypeId)));
+        }
+        return currentEntityType;
+      }).orElseThrow();
+
+    // Build entity type from sources and joins
+//    var source1 = entityType.getSources().get(0);
+//    buildCompositeEntityTypeDefinition(entityType);
+
+    return Optional.of(entityType);
+  }
+
+  private void buildCompositeEntityTypeDefinition(EntityType entityType) {
+    System.out.println("BUILD COMPOSITE DEFINITION");
+    List<EntityTypeSourceJoin> entityTypeSourceJoins = new ArrayList<>();
+    List<EntityTypeSource> sources = entityType.getSources();
+    entityType.setFromClause("BUILD COMPOSITE DEFINITION");
```

Comment:
Personal opinion: we should avoid using the `fromClause` entirely, so that we can remove it from the object model ASAP